### PR TITLE
feat: new StorageAdapter interface

### LINKED ISSUE #3: feat: track chat list
It would be awesome if the plugin could additionally listen for `my_chat_member` updates. That way, it can build up a list of chats the bot is a member of. This is especially useful for broadcasting. (Related: https://github.com/grammyjs/grammY/issues/370)

### PR DESCRIPTION
**Pull request by roziscoding**
This creates the new `StorageAdapter` interface, which uses "cascading keys" in the style used by Deno KV, which allows us to essentially have the idea of a "collection" in the storage adapter.

Now, we can store not only chat members, but also chats and users separately, allowing for a cached `getChat` method, as well as for the existance of a `getUser` method. Besides that, it is now possible to get a list of all known chats, users, and chat members using `get<object>List` functions.

### CODE REVIEW

File: usage.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  for await (const item of itr) {
+    arr.push(item);
+  }
+
+  return arr;
+};
+
+type MyContext = Context & ChatMembersFlavor;
+
+const bot = new Bot<MyContext>("298746736:AAFPI0dkCVA5ieeRQaA78WO4x5vLf5Te3Vc");
```
Comment (roziscoding) [inline usage.ts]: I've revoked this :)

File: src/storage.ts [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+   */
+  values: <V extends T = T>(prefix?: Key) => MaybeAsyncIterable<V>;
+  /**
+   * Lists all key-value pairs via an iterator. Optionally, only lists pairs
+   * with keys that start with a given prefix.
+   */
+  entries: (prefix?: Key) => MaybeAsyncIterable<KV<T>>;
+}
+
+/* ========================= Everything below this should be removed before the v2 release ========================= */
```
Comment (roziscoding) [inline src/storage.ts]: It'll be added to core instead

**Comment by phaalonso** [comment]
There is any change that need to be done to get this merged?

**Comment by KnorpelSenf** [comment]
Unsure. It needs to be tested thoroughly by someone, and then refined as necessary. Could you do that?

**Comment by roziscoding** [comment]
I'll close this for now. In case anyone feels like picking it up, they can just clone the branch.

**Pull request closed without merging**